hudi-io/src/main/java/org/apache/hudi/io/hfile/HFileWriterImpl.java

```diff
@@ -219,6 +219,41 @@ private void initFileInfo() {
         new byte[]{0});
   }
 
+  protected void finishFileInfo() {
```

Comment:
Let's add a functional test that can reproduce the issue, at the HFile reader and writer level i.e., the test fails before the fix and passes after the fix.  We can add HBase as test dependency for such a test on compatibility.